frontend/src/modules/editor/components/FluentEditor.js

```diff
@@ -36,6 +36,35 @@ function annotate(source) {
  * Render an Ace editor for Fluent string editting.
  */
 export default class FluentEditor extends React.Component<EditorProps> {
+    aceEditor: any;
+
+    constructor(props: EditorProps) {
+        super(props);
+        this.aceEditor = React.createRef();
+    }
+
+    componentDidUpdate() {
+        // If there is content to add to the editor, do so, then remove
+        // the content so it isn't added again.
+        // This is an abuse of the redux store, because we want to update
+        // the content differently for each Editor type. Thus each Editor
+        // implements an `updateTranslationSelectionWith` method and that is
+        // used to update the translation.
+        if (this.props.editor.selectionReplacementContent) {
+            this.updateTranslationSelectionWith(
+                this.props.editor.selectionReplacementContent
+            );
+            this.props.resetSelectionContent();
+        }
+    }
```

Comment:
Since the content of this is the same as in `GenericEditor.js`, I wonder if we can factor it out and reuse somehow? Or is that somehow related to the "redux hack"?

Reply:
I've tried to do that but failed. This isn't a case where Higher Order Components work, it requires a class hierarchy, and I'm not too happy about extending classes. Though in this case that might just be the proper solution.